react/features/connection-indicator/statsEmitter.js

```diff
@@ -0,0 +1,174 @@
+import JitsiMeetJS from '../base/lib-jitsi-meet';
+
+declare var APP: Object;
+
+/**
+ * The private value for the local user id. This is used instead of the real id
+ * as the real id might not be set before listeners start subscribing to local
+ * user stat updates. So instead, this constant is used.
+ */
+const LOCAL_USER_ID = 'local';
+
+/**
+ * Contains all the callbacks to be notified when stats are updated.
+ *
+ * {
+ *     userId: Function[]
+ * }
+ */
+const subscribers = {};
+
+/**
+ * A singleton that acts as a pub/sub service for connection stat updates.
+ */
+const statsEmitter = {
+    /**
+     * Emit a stat update to all those listening for local stat updates. Will
+     * also update listeners of remote user stats of changes related to their
+     * stats.
+     *
+     * @param {Object} stats - Connection stats for the local user as provided
+     * by the library.
+     * @returns {void}
+     */
+    localStatsUpdated(stats) {
+        const { framerate, resolution } = stats;
+        const currentLocalUserId = APP.conference.getMyUserId();
+
+        // FIXME overwrites 'lib-jitsi-meet' internal object
+        // Why library internal objects are passed as event's args ?
+        stats.resolution = resolution[currentLocalUserId];
+        stats.framerate = framerate[currentLocalUserId];
+
+        this.remoteStatsUpdated(LOCAL_USER_ID, stats);
+
+        Object.keys(resolution).forEach(id => {
+            if (id === currentLocalUserId) {
+                return;
```

Comment:
(at the very least, if this logic remains, why not use the `APP.conference.isLocalId` call that is used below to test if it's the local user?)